src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+    } else {
+        if (this->path != 0) {
+            this->unk_2A6 = 1;
+
+            if (this->sfxTimer3 == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+                this->sfxTimer3 = 100;
+            }
+
+            this->unk_2A0 = Math_Rand_ZeroFloat(1.99f);
+            this->actor.speedXZ = 4.0f;
+            this->unk_300 = 0.0f;
+            this->unk_2FC = 0.0f;
+            this->actionFunc = func_80AB6A38;
+            return;
+        }
+    }
+
+    iVar1 = 0;
+
+    if (this->timer5 != 0) {
+        if (Math_Rand_ZeroFloat(3.99f) < 1.0f) {
+            this->unk_2E6++;
+            this->unk_2E6 &= 1;
+        }
+
+        Math_SmoothScaleMaxF(&this->unk_26C[9], D_80AB8604[this->unk_2E6], 0.5f, 4000.0f);
+    }
+
+    if (this->timer5 == 0 && this->timer4 == 0) {
+        if (++this->unk_29E >= 8) {
+            this->timer5 = Math_Rand_ZeroFloat(30.0f);
+            this->unk_29E = Math_Rand_ZeroFloat(3.99f);
+
+            if (this->actor.params != 0x000A && this->actor.params != 0x0008) {
+                if (fVar2 < 0.0f) {
+                    fVar2 = fVar2;
+                    fVar2 -= 100.0f;
+                } else {
+                    fVar2 += 100.0f;
+                }
+
+                if (fVar3 < 0.0f) {
+                    fVar3 -= 100.0f;
+                } else {
+                    fVar3 += 100.0f;
+                }
+            } else {
+                fVar2 = Math_Rand_CenteredFloat(30.0f);
+                fVar3 = Math_Rand_CenteredFloat(30.0f);
+
+                if (fVar2 < 0.0f) {
+                    fVar2 -= 20.0f;
+                } else {
+                    fVar2 += 20.0f;
+                }
+
+                if (fVar3 < 0.0f) {
+                    if (1) {} // Required to match
+                    if (1) {}
+                    if (1) {}
+                    if (1) {}
+                    fVar3 -= 20.0f;
+                } else {
+                    fVar3 += 20.0f;
+                }
+            }
+
+            this->unk_2B8 = this->unk_2AC + fVar2;
+            this->unk_2C0 = this->unk_2B4 + fVar3;
+        } else {
+            this->timer4 = 4;
+
+            if (this->actor.bgCheckFlags & 1) {
+                this->actor.speedXZ = 0.0f;
+                this->actor.velocity.y = 3.5f;
+            }
+        }
+    }
+
+    if (this->timer4 != 0) {
+        Math_SmoothDownscaleMaxF(&this->unk_26C[9], 0.5f, 4000.0f);
+        iVar1 = 1;
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.x, this->unk_2B8, 1.0f, this->unk_2FC);
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.z, this->unk_2C0, 1.0f, this->unk_2FC);
+        pi = 0x8000 / M_PI;
+        Math_SmoothScaleMaxF(&this->unk_2FC, 3.0f, 1.0f, 0.3f);
+        fVar2 = this->unk_2B8 - this->actor.posRot.pos.x;
+        fVar3 = this->unk_2C0 - this->actor.posRot.pos.z;
+
+        if (fabsf(fVar2) < 10.0f) {
+            fVar2 = 0.0; // No f is required to match
```

Comment:
I don't think you need comments about no f since it's very obvious the codegen is different between with/without (double to float conversions)

Reply:
Good point, I'll remove them.